plugin/wavesurfer.microphone.js

```diff
@@ -144,6 +149,21 @@ WaveSurfer.Microphone = {
 
         // stop stream from device
         if (this.stream) {
+            var result = this.detectBrowser();
+            // MediaStream.stop is deprecated since:
+            // - Firefox 44 (https://www.fxsitecompat.com/en-US/docs/2015/mediastream-stop-has-been-deprecated/)
+            // - Chrome 45 (https://developers.google.com/web/updates/2015/07/mediastream-deprecations)
+            if ((result.browser === 'chrome' && result.version >= 45) ||
+                (result.browser === 'firefox' && result.version >= 44) ||
+                (result.browser === 'edge')) {
+                if (this.stream.getTracks) { // note that this should not be a call
```

Comment:
Because the method is there in earlier firefox but doesnt fully work until v44 unfortunately